src/convertToObject.js

```diff
@@ -6,7 +6,23 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  const splittedSource = sourceString.split(';');
+
+  for (let i = 0; i < splittedSource.length; i++) {
+    const rule = splittedSource[i].trim();
+
+    if (rule) {
+      const [key, value] = rule.split(':').map((part) => part.trim());
```

Comment:
There's no need to check for 'value !== undefined' since the 'split' method will always return an array with at least one element, even if the string is empty. Simply checking if 'key' is truthy is sufficient.